src/utilities.cpp

```diff
@@ -45,6 +45,21 @@ Eigen::VectorXd DiscreteDifferentiation(const Eigen::VectorXd& input_vector, dou
   return derivative;
 };
 
+Eigen::VectorXd normalize(const Eigen::VectorXd& x)
+{
+  const double min = x.minCoeff();
+  const double max = x.maxCoeff();
+  size_t num_points = x.size();
+  Eigen::VectorXd x_norm = Eigen::RowVectorXd::Zero(num_points);
+
+  for (size_t i = 0; i < num_points; i++)
+  {
+    x_norm(i) = (x(i) - min) / (max - min);
```

Comment:
With Eigen, I bet you could do this for the entire vector at once. Instead of one element at a time